front/components/assistant/conversation/AgentSuggestion.tsx

```diff
@@ -0,0 +1,167 @@
+import { Button, DropdownMenu, RobotIcon } from "@dust-tt/sparkle";
+
+import { useAgentConfigurations } from "@app/lib/swr";
+import { AgentConfigurationType } from "@app/types/assistant/agent";
+import {
+  ConversationType,
+  isAgentMention,
+  isUserMessageType,
+  UserMessageType,
+} from "@app/types/assistant/conversation";
+import { WorkspaceType } from "@app/types/user";
+
+export function AgentSuggestion({
+  owner,
+  userMessage,
+  conversation,
+}: {
+  owner: WorkspaceType;
+  userMessage: UserMessageType;
+  conversation: ConversationType;
+}) {
+  const { agentConfigurations } = useAgentConfigurations({
+    workspaceId: owner.sId,
+  });
+  agentConfigurations.sort((a, b) => compareAgentSuggestions(a, b));
+
+  return (
+    <div className="pt-4">
+      <div className="text-xs font-bold text-element-600">
+        Which KillerZorg would you like to talk with?
+      </div>
+      <div className="mt-3 flex items-center gap-2">
+        {agentConfigurations.slice(0, 3).map((agent) => (
+          <Button
+            key={`message-${userMessage.sId}-suggestion-${agent.sId}`}
+            size="xs"
+            variant="avatar"
+            label={`@${agent.name}`}
+            onClick={() => selectSuggestionHandler(agent)}
+            avatar={agent.pictureUrl}
+          />
+        ))}
+        <DropdownMenu>
+          <DropdownMenu.Button>
+            <Button
+              variant="tertiary"
+              size="xs"
+              icon={RobotIcon}
+              label="Select another"
+            />
+          </DropdownMenu.Button>
+          <div className="relative bottom-6 z-30">
+            <DropdownMenu.Items origin="topLeft" width={320}>
+              {agentConfigurations.slice(3).map((agent) => (
+                <DropdownMenu.Item
+                  key={`message-${userMessage.sId}-suggestion-${agent.sId}`}
+                  label={agent.name}
+                  visual={agent.pictureUrl}
+                />
+              ))}
+            </DropdownMenu.Items>
+          </div>
+        </DropdownMenu>
+      </div>
+    </div>
+  );
+
+  async function selectSuggestionHandler(agent: AgentConfigurationType) {
+    const editedContent = `:mention[${agent.name}]{${agent.sId}} ${userMessage.content}`;
+    await fetch(
+      `/api/w/${owner.sId}/assistant/conversations/${conversation.sId}/messages/${userMessage.sId}/edit`,
```

Comment:
RESTful-ness would call for using `PATCH /api/w/${owner.sId}/assistant/conversations/${conversation.sId}/messages/${userMessage.sId}`  ?

Reply:
A little question though: PATCH suggest we are modifying the resource that the URI points to right?

Here, we are creating a new resource, that will have a different sId and will be referred to by a different URI. Is PATCH still appropriate in that case in your opinion?

Reply:
Given notably that calling GET after the PATCH on the same URI will return the same thing as before the PATCH

Reply:
Fair. Maybe `POST` is better then but fine to keep as is given that it's a bit unclear :+1: